ui/docs/ember-engines.md

```diff
@@ -0,0 +1,158 @@
+
+## Engine’s package.json:
+
+```json
+{
+  "name": "<engine-name>",
+
+  "dependencies": {
+    "ember-cli-htmlbars": "*",
+    "ember-cli-babel": "*"
+  },
+
+  "ember-addon": {
+    "paths": ["../core"]
+  }
+}
+```
+
+For our application, we want to include the **ember-addon** path `../core`
+
+By adding this **ember-addon** path, we are able to share elements between your in-repo addon and the Vault application[^1].
+
+## Configure your Engine
+
+In the engine’s `index.js` file:
+
+```jsx
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+/* eslint-disable node/no-extraneous-require */
+const { buildEngine } = require('ember-engines/lib/engine-addon');
+
+module.exports = buildEngine({
+  name: '<engine-name>',
+  lazyLoading: {
+    enabled: false,
+  },
+  isDevelopingAddon() {
+    return true;
+  },
+});
+```
+
+Within your Engine’s `app/config/environment.js` file:
+
+```jsx
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// config/environment.js
+
+'use strict';
+
+module.exports = function (environment) {
+  const ENV = {
+    modulePrefix: '<engine-name>',
+    environment: environment,
+  };
+
+  return ENV;
+};
+```
+
+Within your Engine’s `addon/engine.js` file:
+
+```jsx
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Engine from '@ember/engine';
+
+import loadInitializers from 'ember-load-initializers';
+import Resolver from 'ember-resolver';
+
+import config from './config/environment';
+
+const { modulePrefix } = config;
+
+export default class <EngineName>Engine extends Engine {
+  modulePrefix = modulePrefix;
+  Resolver = Resolver;
+  dependencies = {
+    services: ['router', 'store', 'secret-mount-path', 'flash-messages'],
+    externalRoutes: ['secrets'],
+  };
+}
+
+loadInitializers(<EngineName>Engine, modulePrefix);
+```
+
+The services in the example above are common services that we often use in our engines. Any other services that are in our main app that is needed for your engine should be added in the dependencies services array.
+
+Optional step: Add some text in the engine’s `application.hbs` file (to see if your engine was set up correctly).
+
+## Register your engine with our main application:
+
+In our `app/app.js` file in the engines object, add your engine’s name and dependencies.
+
+```jsx
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Application from '@ember/application';
+import Resolver from 'ember-resolver';
+import loadInitializers from 'ember-load-initializers';
+import config from 'vault/config/environment';
+
+export default class App extends Application {
+	...
+  engines = {
+    <engine-name>: {
+      dependencies: {
+        services: ['router', 'store', 'secret-mount-path', 'flash-messages', <any-other-dependencies-you-have>],
```

Comment:
I wonder if we could make it clear that services are optional? The suggestions may not necessarily be needed depending on what the engine is used for.